framework/src/controller/bus.ts

```diff
@@ -225,18 +236,29 @@ export class Bus {
 		});
 	}
 
-	public publish(eventName: string, eventValue?: object): void {
+	public publish(eventData: string | JSONRPC.NotificationRequest): void {
+		const parsedEvent = Event.fromJSONRPCNotification(eventData);
+
+		try {
+			JSONRPC.validateJSONRPCRequest(parsedEvent.toJSONRPCNotification() as never);
+		} catch (error) {
+			throw JSONRPC.errorResponse(null, JSONRPC.invalidRequest());
```

Comment:
i just realized that this is not `Error` instance. when it throws it should be always error instance.

Reply:
I missed it as it was not highlighted by eslint.